app/plugin.go

```diff
@@ -747,56 +760,65 @@ func (a *App) processPrepackagedPlugins(pluginsDir string) []*plugin.Prepackaged
 
 	pluginSignaturePathMap := getPluginsFromFilePaths(fileStorePaths)
 	plugins := make([]*plugin.PrepackagedPlugin, 0, len(pluginSignaturePathMap))
+	prepackagedPlugins := make(chan plugin.PrepackagedPluginResult, len(pluginSignaturePathMap))
 	for _, pluginPaths := range pluginSignaturePathMap {
-		plugin, err := a.processPrepackagedPlugin(pluginPaths)
-		if err != nil {
-			mlog.Error("Failed to install prepackaged plugin", mlog.String("path", pluginPaths.path), mlog.Err(err))
+		go a.processPrepackagedPlugin(pluginPaths, prepackagedPlugins)
+	}
+
+	for i := 0; i < len(pluginSignaturePathMap); i++ {
+		result := <-prepackagedPlugins
+		if result.Err != nil {
+			mlog.Error("Failed to install prepackaged plugin", mlog.String("path", result.Path), mlog.Err(err))
 			continue
 		}
 
-		plugins = append(plugins, plugin)
+		plugins = append(plugins, result.P)
 	}
-
 	return plugins
 }
 
 // processPrepackagedPlugin will return the prepackaged plugin metadata and will also
 // install the prepackaged plugin if it had been previously enabled and AutomaticPrepackagedPlugins is true.
```

Comment:
I think the documentation for this function should reflect that we return the data through a channel and not directly.